lib/ansible/modules/database/postgresql/postgresql_query.py

```diff
@@ -0,0 +1,347 @@
+        kw["host"] = module.params["login_unix_socket"]
+
+    try:
+        pgutils.ensure_libs(sslrootcert=module.params.get('ssl_rootcert'))
+        db_connection = psycopg2.connect(**kw)
+
+        # Enable autocommit on demand only so we can create databases
+        if module.params['autocommit']:
+            if psycopg2.__version__ >= '2.4.2':
+                db_connection.autocommit = True
+            else:
+                db_connection.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
+
+        # Using RealDictCursor allows access to row results by real column name
+        cursor = db_connection.cursor(cursor_factory=psycopg2.extras.RealDictCursor)
+
+    except TypeError:
+        e = get_exception()
+        if 'sslrootcert' in e.args[0]:
+            module.fail_json(msg='''Postgresql server must be at least version
+                             8.4 to support sslrootcert.
+                             Exception: {0}'''.format(e), exception=traceback.format_exc())
+        module.fail_json(msg="unable to connect to database: %s" % e, exception=traceback.format_exc())
+
+    except Exception:
+        e = get_exception()
+        module.fail_json(msg="unable to connect to database: {0}".format(str(e)), exception=traceback.format_exc())
+
+    # if query is a file, load the file and run it
+    query = module.params["query"]
+    if query.endswith('.sql'):
+        try:
+            query = open(query, 'r').read().strip('\n')
+        except Exception:
+            e = get_exception()
+            module.fail_json(msg="Unable to find '%s' in given path." % query)
+
+    arguments = None
+
+    # prepare args
+    if module.params["positional_args"] is not None:
+        arguments = module.params["positional_args"]
+
+    elif module.params["named_args"] is not None:
+        arguments = module.params["named_args"]
+
+    try:
+        cursor.execute(query, arguments)
+    except Exception:
+        e = get_exception()
+        module.fail_json(msg="Unable to execute query: %s" % e,
+                         query_arguments=arguments)
+
+    ansible_facts = {}
+    query_results = []
+    if cursor.rowcount > 0:
+        # There's no good way to return results arbitrarily without inspecting
+        # the SQL, so we act consistent and return the empty set when there's
+        # nothing to return.
+        try:
+            query_results = cursor.fetchall()
+        except psycopg2.ProgrammingError:
+            pass
+
+        rowcount = len(query_results)
+        fact = module.params["fact"]
+        if fact is not None:
+            ansible_facts = {fact: query_results}
+    else:
+        rowcount = 0
+
+    statusmessage = cursor.statusmessage
+
+    changed = False
+
+    # set changed flag only on non read-only command
+    if ("UPDATE" in statusmessage or "INSERT" in statusmessage or
+            "UPSERT" in statusmessage or "DELETE" in statusmessage):
+        changed = True
```

Comment:
What about `MOVE`, `COPY`, `CREATE` ? It doesn't seem reliable, `changed` should be `True` if the query has been executed. By the way I am not sure if supporting `check_mode` is a good idea. `command` and `shell` modules don't support `check_mode`, it would be fine it this module didn't support `check_mode`.